esx_service/cli/vmdkops_admin_test.py

```diff
@@ -780,28 +799,63 @@ def test_tenant_access(self):
+                                                    default_datastore=auth_data_const.VM_DS,
                                                     description="Test tenant1",
                                                     vm_list=[self.vm1_name],
                                                     privileges=[])
         self.assertEqual(None, error_info)
 
-        # add first access privilege for tenant
+        # now, should only have privilege to ""__VM_DS
+        error_info, privileges = auth_api._tenant_access_ls(self.tenant1_name)
+        rows = vmdkops_admin.generate_tenant_access_ls_rows(privileges)
+        self.assertEqual(1, len(rows))
+        expected_output = [auth_data_const.VM_DS,
+                            "True",
+                            "Unset",
+                            "Unset"]
+        actual_output = [rows[0][0],
+                         rows[0][1],
+                         rows[0][2],
+                         rows[0][3]]
+        self.assertEqual(expected_output, actual_output)
+
+        # remove the privilege to "__VM_DS", which should fail
+        # since "__VM_DS still the "default_datastore" for tenant1
+        error_info = auth_api._tenant_access_rm(name=self.tenant1_name,
+                                                datastore=auth_data_const.VM_DS
+                                                )
+        self.assertNotEqual(None, error_info)
+
+        # add a access privilege for tenant to self.datastore_name
         # allow_create = False
         # max_volume size = 600MB
         # total_volume size = 1GB
         volume_maxsize_in_MB = convert.convert_to_MB("600MB")
         volume_totalsize_in_MB = convert.convert_to_MB("1GB")
         error_info = auth_api._tenant_access_add(name=self.tenant1_name,
                                                  datastore=self.datastore_name,
-                                                 default_datastore=False,
                                                  allow_create=False,
                                                  volume_maxsize_in_MB=volume_maxsize_in_MB,
                                                  volume_totalsize_in_MB=volume_totalsize_in_MB
                                              )
         self.assertEqual(None, error_info)
 
+        # update the "default_datastore" to self.datastore_name
+        error_info = auth_api._tenant_update(name=self.tenant1_name,
+                                             default_datastore=self.datastore_name)
+        self.assertEqual(None, error_info)
+
+        # try to remove the privilege to "__VM_DS" again, which should not fail
+        # since the "default_datastore" for tenant1 is set to self.datastore_name
+        error_info = auth_api._tenant_access_rm(name=self.tenant1_name,
+                                                datastore=auth_data_const.VM_DS
+                                                )
+        self.assertEqual(None, error_info)
+
```

Comment:
Could add a test to create vmgroup with VM_DS as the default DS and add VMs from at least two different DS'es to the vmgroup and verify volumes created from each VM are created in the respective datastores.